### ISSUE TITLE
Submission of scDiagnostics to Bioconductor

### ISSUE THREAD

**Issue by AnthonyChristidis** [Bioconductor/Contributions#3449]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/ccb-hms/scDiagnostics

Confirm the following by editing each check box to '[x]'

- [x ] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x ] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x ] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x ] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x ] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x ] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x ] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [ x] The 'devel' branch for new packages and features.
- [ x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x ] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by lgeistlinger** [comment]
Hi we would like to request that this submission is assigned to @HelenaLC as reviewer, if possible, given that she is an expert in single-cell analysis and has written diagnostic functionality for CyTOF data that seems highly relevant here.

**Comment by lgeistlinger** [comment]
Also: this package has been accepted for a [package demo](https://cfp.bioconductor.org/2024/talk/SYXCHV/) at the upcoming Bioc2024 conference. We understand that there are several constraints to the review process, but it would be great if we would be able to proceed with the review process in a timely manner so that the package would be available from Bioconductor in time for the package demo at the conference.

**Comment by lshep** [comment]
I'm getting an ERROR trying to R CMD build ...  

```
 4: do.call(FUN, c(list(x = x, k = k, nu = nu, nv = nv, center = center,     scale = scale, BPPARAM = BPPARAM, ...), ARGS(BSPARAM)))
 5: runSVD(x, k = k, nu = nu, nv = nv, center = center, scale = scale,     BPPARAM = BPPARAM, ..., BSPARAM = BSPARAMFUN(deferred = bsdeferred(BSPARAM),         fold = bsfold(BSPARAM)))
 6: runSVD(x, k = k, nu = nu, nv = nv, center = center, scale = scale,     BPPARAM = BPPARAM, ..., BSPARAM = BSPARAMFUN(deferred = bsdeferred(BSPARAM),         fold = bsfold(BSPARAM)))
 7: runSVD(x, k = rank, nu = ifelse(get.pcs, rank, 0), nv = ifelse(get.rotation,     rank, 0), center = center, scale = scale, ...)
 8: runSVD(x, k = rank, nu = ifelse(get.pcs, rank, 0), nv = ifelse(get.rotation,     rank, 0), center = center, scale = scale, ...)
 9: .local(x, ...)
10: runPCA(x, rank = ncomponents, BSPARAM = BSPARAM, BPPARAM = BPPARAM)
11: runPCA(x, rank = ncomponents, BSPARAM = BSPARAM, BPPARAM = BPPARAM)
12: .calculate_pca(mat, transposed = !is.null(dimred), ...)
13: .local(x, ...)
14: calculatePCA(y, ...)
15: calculatePCA(y, ...)
16: .local(x, ...)
17: runPCA(ref_data)
18: runPCA(ref_data)
19: eval(expr, envir, enclos)
20: eval(expr, envir, enclos)
21: eval_with_user_handlers(expr, envir, enclos, user_handlers)
22: withVisible(eval_with_user_handlers(expr, envir, enclos, user_handlers))
23: withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)
24: handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler))
25: timing_fn(handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)))
26: evaluate_call(expr, parsed$src[[i]], envir = envir, enclos = enclos,     debug = debug, last = i == length(out), use_try = stop_on_error !=         2L, keep_warning = keep_warning, keep_message = keep_message,     log_echo = log_echo, log_warning = log_warning, output_handler = output_handler,     include_timing = include_timing)
27: evaluate::evaluate(...)
28: evaluate(code, envir = env, new_device = FALSE, keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error && options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options))
29: in_dir(input_dir(), expr)
30: in_input_dir(evaluate(code, envir = env, new_device = FALSE,     keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error && options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options)))
31: eng_r(options)
32: block_exec(params)
33: call_block(x)
34: process_group(group)
35: withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e))
36: xfun:::handle_error(withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e)),     function(e, loc) {        setwd(wd)        write_utf8(res, output %n% stdout())        paste0("\nQuitting from lines ", loc)    }, if (labels[i] != "") sprintf(" [%s]", labels[i]), get_loc)
37: process_file(text, output)
38: knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
39: rmarkdown::render(file, encoding = encoding, quiet = quiet, envir = globalenv(),     output_dir = getwd(), ...)
40: vweave_rmarkdown(...)
41: engine$weave(file, quiet = quiet, encoding = enc)
42: doTryCatch(return(expr), name, parentenv, handler)
43: tryCatchOne(expr, names, parentenv, handlers[[1L]])
44: tryCatchList(expr, classes, parentenv, handlers)
45: tryCatch({    engine$weave(file, quiet = quiet, encoding = enc)    setwd(startdir)    output <- find_vignette_product(name, by = "weave", engine = engine)    if (!have.makefile && vignette_is_tex(output)) {        texi2pdf(file = output, clean = FALSE, quiet = quiet)        output <- find_vignette_product(name, by = "texi2pdf",             engine = engine)    }    outputs <- c(outputs, output)}, error = function(e) {    thisOK <<- FALSE    fails <<- c(fails, file)    message(gettextf("Error: processing vignette '%s' failed with diagnostics:\n%s",         file, conditionMessage(e)))})
46: tools::buildVignettes(dir = ".", tangle = TRUE)
An irrecoverable exception occurred. R is aborting now ...
Segmentation fault (core dumped)
```

**Comment by vjcitn** [comment]
I was able to build with R 4.4, Bioc 3.19.  Will try with 3.20 container.

**Comment by vjcitn** [comment]
vignette has eval=FALSE for all chunks I looked at, so figures are not produced ...

**Comment by HelenaLC** [comment]
Hi there, glad to see this coming through; here my first impressions. While these are mostly stylistic comments and should be simple enough to address, I feel the "crucial" bit is simplifying examples (see below). I think these should let the user have a quick peak at how to execute the function, what it returns etc., but they currently take quite some time to execute, and require installing most "suggested" packages... the vignette can be used for a more lengthly (biologically meaningful) demo, which it does nicely already. Please report back with what has/not been address how/why not, and any questions you might have; thanks!


# misc

- [ ] please consider implementing unit test; we strongly encourage them!

- [ ] consider adding a package help page (accessible via `?scDiagnostics`) with an overview of key functionality and links to main functions

- [ ] there are some inconsistencies in vertical code alignment vs. 4-space tab indentation; this is generally fine, however, to ease readability, it'd be nice to stick to an 80-character limit in examples especially, specifically when variable & function names are long (e.g., the `calculateAveragePairwiseCorrelation` example doesn't show with my standard RStudio panel sizing)

- [ ] it would be nice to start out with a clean-ish CHECK/BiocCheck report (tends to only get worse over time...); in most cases, R CMD CHECK NOTES stating "no visible binding" can be addressed by using `aes(.data$x, ...)` (instead of `aes(x)`) and adding `@importFrom ggplot2 .data`

# code

- [ ] many examples have quite long runtimes (>5s & up to 1min), largely attributable to loading the data (from online) and some heavy-ish computations; perhaps worth considering adding example data under `inst/extdata`, alongside a piece of documentation and a script under `inst/scripts` that recapitulates how that was retrieved/subsetted/processed? I personally find it quite tedious to have to download something and wait many seconds to see how a function works/what it returns (unless this is really necessary)...  e.g., stashing a reference & query scRNA-seq data (subset) and corresponding `SingleR` results would suffice for most examples, I think? And then, perhaps, only showing the full workflow (normalization, transfer, selection, reduction, ...) for selected examples but not, say, every visualization? Importantly, this would also help to not require users to install various `Suggests:` to run examples (c.f. `scuttle/scater` examples which rely on toy data mostly)

- [ ] avoid `1:...`; use `seq_len/along()` instead

- [ ] not sure where all this applies, but when matching an argument against a predefined set of character strings (e.g., `plot.plotGeneSetScores.R`, `projectPCA.R` etc.), code can be simplified using `match.arg()` and (optionally) `switch()`, e.g.:

```
foo <- \(se_object, method=c("PCA", "TSNE", "UMAP")) {
    arg <- match.arg(arg) # aleady checks validity
    mtx <- switch(arg, 
        PCA={...}, # do something special
        reducedDim(obj, arg) # otherwise just this
    })
    ...
}
```

- [ ] not sure where/if this would break any function, but maybe worth adding a more validity checks on "simple" function arguments everywhere (e.g., is scalar, is logical/character/numeric etc.); e.g., passing 2 values to `xxx_col` in `plotQCvsAnnotation()` would break line 114 and the plotting due to renaming at line 117 

- [ ] fwiw...

```
# plotQCvsAnnotation.R:
qc_stats <- colData(query_data)[, qc_col]
cell_type_scores <- colData(query_data)[, score_col]
cell_labels <- colData(query_data)[[label_col]]
data <- data.frame(
    QCStats = qc_stats,
    Scores = cell_type_scores, 
    CellType = cell_labels)

# can be written as...
qc_stats <- query_data[[qc_col]]
cell_type_scores <- query_data[[score_col]] 
cell_labels <- query_data[[label_col]] 
data <- ...

# or... 
cols <- c(qc_col, score_col, label_col)
data <- colData(query_data)[, cols]
names(data) <- c("QCStats", "Scores", "CellType")
```

- preallocate-and-fill should try and use `v/m/lapply()` rather than a `for`-loop; e.g., `calculateHotellingPValue()`:

```
# Create a list to store p-values for each cell type
p_values <- rep(NA, length(unique_cell_types))
names(p_values) <- unique_cell_types
    
for (cell_type in unique_cell_types) {
    # Subset principal component scores for current cell type
    ref_subset_scores <- pca_output$ref[which(cell_type == reference_data[[ref_cell_type_col]]), pc_subset]
    query_subset_scores <- pca_output$query[which(cell_type == query_data[[query_cell_type_col]]), pc_subset]
    # Calculate the p-value
    hotelling_output <- Hotelling::hotelling.test(x = ref_subset_scores, y = query_subset_scores)
    # Store the result
    p_values[cell_type] <- hotelling_output$pval
}

# can be written as... (mock code)

names(unique_cell_types) <- unique_cell_types
idx_ref <- split(ncol(ref), ref[[ref_typ_col]])
idx_que <- split(ncol(que), que[[que_typ_col]])

p_values <- vapply(unique_cell_types, \(.) {
    ref_sco <- pca_out$ref[idx_ref[[.]], pc_subset]
    que_sco <- pca_out$que[idx_que[[.]], pc_subset]
    hotelling_output <- Hotelling::hotelling.test(x=ref_sco, y=que_sco)
    return(hotelling_output$pval)
}, numeric(1))
```

# docs

- [ ] repeating affiliations can be implemented as:

```
author:
- name: Peter Pan
  affiliation:
  - &foo Neverland
- name: Captain Hook
  affiliation:
  - *foo
```

- [ ] please replace GitHub by normal Bioc installation instructions, i.e., `BiocManager::install("ccb-hms/scDiagnostics")` > `"scDiagnostics"` (also in the README!)

- [ ] code chunk at lines 320+ is largely commented out code; also line 842? Commented out code should be removed for release

The following are mostly stylistic comments, which I'm aware are not strictly required; however, I feel that proper usage of text-style options and code formatting aren't negligible for legibility...

- [ ] you can hyperlink external packages using `BiocStyle` functions, e.g., "\[remotes\]\(https://...\)" > "\`r BiocStyle::CRANpkg("remotes")\`" and "AUCell" > "\`r BiocStyle::Biocpkg("AUCell")\`"

- [ ] "scDiagnostics" (the package) is sometimes in code-style and sometimes in plain text; be consistent

- [ ] when introducing example datasets (lines 102+ and 165+), perhaps worth hyperlinking to the source and/or DOI and/or adding proper citations (@AuthorYear), to make things easier for the reader?

- [ ] to ease readability, I'd suggests using code-style (and maybe parentheses) to distinguish variable/class/slot/function/argument etc. names from plain text; e.g., "colData of" > `colData` "plotGeneExpressionDimred" > `plotGeneExpressionDimred()`

- [ ] for readability, especially in the vignette, I'd strongly recommend sticking to the recommended 80-character limit and refraining from vertical code alignment (using 4-space tab indentation instead), as the latter introduces a lot of superfluous white space, especially when variable and function names are lengthy; note that this will mostly be displayed online (in the browser), so it's nice to not require horizontal scrolling (e.g., code chunk at lines 434+ is hard to parse)

**Comment by AnthonyChristidis** [comment]
Thank you so much, @HelenaLC, for your prompt and thorough review - these comments are very useful and addressing them will only make the package better. 

I was already aware of some of the issues you brought up and had been working on them for a while, so I hope to be able to submit the update shortly with all these points addressed.

I will let you know if there is anything I would like to consult with you before submitting the updated package for your follow-up review.

**Comment by HelenaLC** [comment]
Any update on this? Thanks.

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC, thanks for reaching out. I recently came back from the BioC2024 conference in Grand Rapids (MI). Ludwig and I will meet shortly to discuss some final touch-ups based on the feedback we received before re-submitting the package for final review.

From my end, I incorporated all of your feedback and improved the functionality of the package further. I am also in the process of finalizing the multiple vignettes for the package (one "Get Started" and a few specialized vignettes that bunches functions with a similar purpose together).

I expect that I will submit the revised version of the package next week by Friday latest. I will keep in touch and notify you when I submit to provide some specifications/details of the revised submission.

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC. Just letting you know I will be submitting next week, by Friday latest. Sorry for the delay and thanks for your patience on this!

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC, I'm glad to finally announce that I submitted the revised version of scDiagnostics, in which I addressed all of your comments to the best of my knowledge.

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC , I'm glad to finally announce that I submitted the revised version of scDiagnostics, in which I addressed all of your comments to the best of my knowledge.

**misc**:
- I have now added datasets to the package that have been curated/processed and the user can now use them to test the functions in a much more accessible manner (they already contain all the data/metadata to run the examples, i.e. the reducedDims data, the colData, etc). So this should also have simplified the examples for each function by reducing the  preamble code of the examples.
- I have implemented unit tests for every single exported function in the package, including the S3 plot methods.
- I have added a ?scDiagnostics help page which now contains an overview of key functionality and also some links to main functions.
- I have fixed most of the 4-spac tab indentation issues, although some remain.
- I fixed all "no visible binding" issues through the use of the "rlang" package.

**code**:
- I have fixed the code examples such that all are executed below 10 seconds on any reasonable machine.
- I have removed the use of 1:... whenever possible. Some remain where the code would be odd if seq_len()/along() would be used.
- I have used match.arg() whenever possible to match an argument against a predefined set of character string.
- On top of adding more validity checks, I have created an internal argumentCheck() function that is recycled for almost every function in the package to check every single argument passed by the user.

**docs**:
- Please see the [pkgdown website](https://ccb-hms.github.io/scDiagnostics/) for an overview of the package docs.
- For now, since scDiagnostics is not on Bioconductor yet, I still recommend to install the package from the ccb-hms GitHub repository, but as soon as you let me know I can change it to Bioconductor, I will do so.
- Note that there are now seven vignettes, including one "Getting Started with scDiagnostics". Each of the other vignettes bunch the functions together around a common theme. In the pkgdown website I also organized the functions in the [Reference](https://ccb-hms.github.io/scDiagnostics/) tab by common purpose/theme.
- I added hyperlinks whenever possible throughout the documentation.
- I added citations for the documentation of the datasets, also also references/citations for the key statistical methods/implementations used throughout the package.
- I incorporated BioC-style code as much as possible, including style to distinguish variables, classes, functions, slots, etc.
- As mentioned I tried to fix the 80-character limit for each line as much as possible. There are still some improvements that can be made, for for the sake of not making some of the functions too long, I kept some of these long lines as is (but let me know if you would still prefer to prioritize line length over function length).

Thanks again for your thorough review and all your help on this. I look forward to your feedback and eventual publication of this work on Bioconductor.

**Comment by HelenaLC** [comment]
Thanks Anthony, it's come along nicely -- build report is much cleaner, very much appreciate putting validity checks & unit tests in place, and examples run a whole lot better (with relevant data in place) -- couple minor points below ... also, not sure I'm missing something, but I don't see any of my previous comments under **docs** having been addressed? E.g., link to datasets, `BiocStyle`, affiliations -- all still looks much the same to me?

- [ ] `data-raw` scripts should go under `inst/script/` according to guidelines

- [ ] re `seq_len/along()` -- I see your point with `1:10` (fixed number), but something like `1:n_top_vars` should use `seq_len()`; please double check for such cases (package-wide search for "1:[a-z]" should do the trick)

- [ ] re `match.arg()` -- you can avoid `if (!x %in% c(...))`, this is already checked for (e.g., `plotGeneSetScores() lines 51+`) ... also here, for dimension reductions, might be worth adding one check that it's in `reducedDimNames()` upfront rather than 3 independent checks & stop messages...

- (this isn't a real review comment, rather perhaps helping with better coding practice in the future :) to given one example, in `calculateHotellingPValue()`, when you repeat the same (long) sequence of characters > 2 times, code is *a lot* less cluttered using an intermediate variable... e.g.,

```
observed_t2 <- hotellingT2(
  as.matrix(cell_list[[cell_type]][ref_ind, paste0("PC", pc_subset)]),
  as.matrix(cell_list[[cell_type]][!ref_ind, paste0("PC", pc_subset)]))
...
perm_t2[perm_id] <- hotellingT2(
  as.matrix(cell_list[[cell_type]][ref_sample_id, paste0("PC", pc_subset)]),
  as.matrix(cell_list[[cell_type]][-ref_sample_id, paste0("PC", pc_subset)]))

# could be:
x <- as.matrix(cell_list[[cell_type]][, paste0("PC", pc_subset)])
observed_t2 <- hotellingT2(x[ref_ind, ], x[!ref_ind, ])
...
perm_t2[perm_id] <- hotellingT2(x[ref_sample_id, ], x[-ref_sample_id, ])
```

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC, thanks a lot for your thorough review once again and your clear comments. Apologies that some of the formatting **doc** comments were not updated as per your request. I had incorporated these formatting suggestions but accidentally reversed the in a later update to the package.

I will incorporate all your remaining suggestions on top of the **doc** formatting comments, and will resubmit by Monday next week. I will ping you with some comments when I resubmit.

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC, just letting you know that I incorporated all of your feedback (and beyond) in the version I just submitted. Note that the affiliations follow the BiocStyle that you requested in the vignettes, but it is not reflected in the pgkdown website for some reason.

I look forward to any remaining feedback, and to have the package published on the Bioconductor platform.

**Comment by HelenaLC** [comment]
Thanks again Anthony. I'm happy to accept in principal, would only like to resolve 1 outstanding build report NOTE on `LadyData: true` (see note [here](https://contributions.bioconductor.org/description.html?q=lazy#description-lazydata)); could you either justify this or else set it to `false`?

Previous suggestions on `calculateHotellingPValue()` still hold; and, in `plotGeneSetScores()` (or similar code structures) you could write `if (method == "PCA") { ... } else { plot_mat <- reducedDim(se_object, method) }`. Both of these are not real reviews but personal suggestions for code clarify/brevity, so no must by any means.

**Comment by AnthonyChristidis** [comment]
Hi @HelenaLC, I updated the package to incorporate all your final comments (including your non-mandatory suggestions for coding style). I also did some other minor formatting/style updates following some comments from Ludwig.

Note that in this version I also added one final set of functions in case you want to review it (it follows the same coding style as other functions, so perhaps it is not necessary for you to look at). They are projectSIR, calculateSIRSpace and the S3 method for calculateSIRSpace.

I bumped the version on the ccb-hms GitHub repository to 1.0.0 as I suppose this should be the initial version ID for Bioconductor.

Thanks again for your patience and thorough review for this rather massive package. I've previously published over 13 CRAN packages but this was my first one on the Bioconductor platform, and all your feedback was extremely valuable and helpful.

**Comment by HelenaLC** [comment]
I don't see changes re code-style, but yeah, these aren't necessary - see NOTE in the report that you should drop `LazyDataCompression:` in the DESCRIPTION when `LadyData` isn't `true` - I'll accept now in any case. Thanks a million for all your efforts and the pleasant communication throughout!

**Comment by AnthonyChristidis** [comment]
Thanks, @HelenaLC. It's been great to work on this with you. I may have another Bioconductor package coming in before the end of the year (it won't be nearly as massive as this one, though). Maybe we'll go at this once again in the not-so-distant future. =)

For the coding style comments, I did incorporate the changes for `calculateHotellingPValue()` and `plotGeneSetScores()`. Unless you were talking about something else? Although I'm sure there are other areas where similar changes could be made.

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/AnthonyChristidis.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("scDiagnostics")`. The package 'landing page' will be created at

https://bioconductor.org/packages/scDiagnostics

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
